lte/gateway/c/session_manager/SessionStateEnforcer.cpp

```diff
@@ -302,26 +372,67 @@ void SessionStateEnforcer::m5g_complete_termination(
   return;
 }
 
-/*Removing all associated rules to session*/
-void SessionStateEnforcer::m5g_remove_rules_and_update_upf(
-    const std::string& imsi, const std::unique_ptr<SessionState>& session,
-    const std::string& dnn, SessionStateUpdateCriteria& uc) {
-  // remove all rules;
-  MLOG(MINFO) << "Will be removing all associated rules of session id "
+void SessionStateEnforcer::m5g_state_change_action(
+    std::string& imsi, SessionStateUpdateCriteria& session_uc,

```

Comment:
Another nit, but we've been putting update related input parameters at the end for all functions. Can we follow the pattern here as well?
```
void SessionStateEnforcer::m5g_state_change_action(
    std::string& imsi, std::unique_ptr<SessionState>& session, 
    SetSmNotificationContext notif, SessionStateUpdateCriteria& session_uc,)
```

Reply:
adopted.